hack/test_e2e.sh

```diff
@@ -535,25 +536,41 @@ function help {
     #   None                                                                  #
     ###########################################################################
     printf "\n"
-    printf "Usage: %s [-i ip_family] [-b backend]\n" "$0"
+    printf "Usage: %s [-i ip_family] [-b backend] [-n CNI name]\n" "$0"
     printf "\t-i set ip_family(ipv4/ipv6/dual) name in the e2e test runs.\n"
     printf "\t-b set backend (iptables/nft/ipvs) name in the e2e test runs.\n"
-    printf "\t-c flag allows for ci_mode. Please don't run on local systems. \n"
-    printf "\nExample:\n\t %s -i ipv4 -b iptables\n" "${0}"
+    printf "\t-c flag allows for ci_mode. Please don't run on local systems.\n"
+    printf "\t-d devel mode, creates the test env but skip e2e tests. Useful for debugging.\n"
+    printf "\t-n CNI name (kindnet(default)/calico/cilium/antrea/ovnkubernetes).\n"
+    printf "\nExample:\n\t %s -i ipv4 -b iptables -n kindnet\n" "${0}"
     exit 1 # Exit script after printing help
 }
 
 ci_mode=false
-while getopts "i:b:c" flag
+devel_mode=false
+cni_selected=false
+while getopts "i:b:cdn:" flag
 do
     case "${flag}" in
         i ) ip_family="${OPTARG}" ;;
         b ) backend="${OPTARG}" ;;
         c ) ci_mode=true ;;
+        d ) devel_mode=true ;;
+        n ) cni_selected="${OPTARG}" ;;
         ? ) help ;; #Print help
     esac
 done
 
+if [[ "${cni_selected}" = false ]]; then
+    cni_selected="kindnet"
+fi
+
+if ! [[ "${cni_selected}" =~ ^(kindnet|calico|cilium|antrea|ovnkubernetes)$ ]]; then
```

Comment:
to avoid confusion can we add only the ones we support and go incrementing according

Reply:
sure